modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/application/ApplicationDescriptorFactoryTestCase.java

```diff
@@ -84,11 +84,16 @@ private static File getResourceFile(String resource) {
   public static final String JAR_FILE_NAME = "test.jar";
 
   @Rule
-  public SystemProperty remoteMavenRepository = new SystemProperty("muleRuntimeConfig.maven.repositoryLocation",
-                                                                   discoverProvider(ApplicationDescriptorFactoryTestCase.class
-                                                                       .getClassLoader()).getLocalRepositorySuppliers()
-                                                                           .environmentMavenRepositorySupplier().get()
-                                                                           .getAbsolutePath());
+  public SystemProperty repositoryLocation = new SystemProperty("muleRuntimeConfig.maven.repositoryLocation",
+                                                                discoverProvider(ApplicationDescriptorFactoryTestCase.class
+                                                                    .getClassLoader()).getLocalRepositorySuppliers()
+                                                                        .environmentMavenRepositorySupplier().get()
+                                                                        .getAbsolutePath());
+
+  @Rule
+  public SystemProperty remoteMavenRepository =
```

Comment:
I think that there is another problem here, as the code before this change was always getting the artifacts resolved from the local repository. Where there is a logic to resolve it by searching first for the "localRepository" system property (set by Maven) or https://github.com/mulesoft/mule-maven-client/blob/master/mule-maven-client-impl/src/main/java/org/mule/maven/client/internal/DefaultLocalRepositorySupplierFactory.java#L69
I'm wondering if this test was failing as it was trying to resolve a plugin that was not build yet (an extension for instance) but as it is deployment module it cannot depend on it.
If that's the case we should create a test extension/plugin on this module to test this logic instead of depending on a real one.
I just found that this test is trying to validate an scenario where the application descriptor to be created has http extension.
This is a problem as it will never be built before this module is build.
I will keep this "fix" and create another task to remove these kind of tests to another module.